code/game/objects/structures/popout_cake.dm

```diff
@@ -0,0 +1,156 @@
+//Pop-out cake!
+//A huge cake that can fit a person inside
+//You can cut the cake up to 16 times to receive a cake slice. After that the cake is destroyed and a large cardboard box is left
+
+//LOCKING CATEGORY FOR OBJECTS THAT ARE CLIMBING OUT OF THE CAKE
+/datum/locking_category/popout_cake
+	pixel_y_offset = 1 //First row of pixels on the cake is transparent :^
+
+	var/pixel_y_offset_max = 24
+
+/datum/locking_category/popout_cake/lock(atom/movable/AM)
+	AM.pixel_y = 0
```

Comment:
but setting pixel_x/y is kinda necessary here, otherwise the animation looks bad